### PR TITLE
Build & attach release assets automatically

### PR DESCRIPTION
**Pull request by MaxHorstmann**
A workflow which automatically builds and attaches binaries to a release when it's published.

image

### CODE REVIEW

**Comment by MaxHorstmann** [comment]
@tlimoncelli let's chat about this when you're back!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.15.3
```
Comment (atma-stackoverflow) [inline .github/workflows/release.yml:26]: For this, I think "at least 1.15" is sufficient; i.e. `^1.15`.  If .3 point release is required to build then what you have is appropriate.  If you want it locked to EXACTLY 1.15.3, it'd be `1.15.3`

Comment (MaxHorstmann) [reply .github/workflows/release.yml]: Agreed! https://github.com/StackExchange/dnscontrol/commit/bf7cb5bb03117a5d879383f4eab29564dc8c0f1a

**Comment by MaxHorstmann** [comment]
Tested this one more time on master branch, works as expected:

image

[Release Docs](https://github.com/StackExchange/dnscontrol/blob/master/docs/release-engineering.md#step-6-publish-the-release) are updated to reflect the change.

@tlimoncelli @atma-stackoverflow

**Pull request merged**